modules/javafx.web/src/test/java/test/javafx/scene/web/PointerEventsTest.java

```diff
@@ -0,0 +1,233 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.javafx.scene.web;
+
+import java.io.File;
+import static java.util.Arrays.asList;
+import java.util.concurrent.CountDownLatch;
+import javafx.event.EventType;
+import javafx.scene.input.MouseButton;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.web.WebView;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.application.Platform;
+import org.junit.BeforeClass;
+import com.sun.javafx.application.PlatformImpl;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import javafx.concurrent.Worker.State;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+public class PointerEventsTest {
+
+    @BeforeClass
+    public static void setupOnce() {
+        final CountDownLatch startupLatch = new CountDownLatch(1);
+
+        PlatformImpl.startup(() -> {
+            startupLatch.countDown();
+        });
+
+        try {
+            startupLatch.await();
+        } catch (InterruptedException ex) {
+        }
+    }
+
+    /**
+     * This test loads a website with SVG shapes "polyline", "path", "rect",
+     * "circle", "ellipse", "polygon". By setting the CSS property
+     * "pointer-events" to "stroke", the rendering engine is configured to only
+     * react to clicks on the stroke of the shapes.
+     *
+     * <p>With a javascript event click handler, mouse events on the shapes
+     * are recorded.</p>
+     *
+     * <p>This tests simulates clicks on the stroke, so the shapes should have
+     * received clicks and report being activated.</p>
+     */
+    @Test
+    public void testClickOnStrokePointerEventsStroke() throws Exception {
+        WebView[] viewHolder = new WebView[1];
+        CountDownLatch waitTestDocumentLoaded = new CountDownLatch(1);
+
+        Platform.runLater(() -> {
+            final Stage primaryStage = new Stage();
+            final WebView view = new WebView();
+            viewHolder[0] = view;
+            primaryStage.setScene(new Scene(view, 800, 600));
+            primaryStage.show();
+
+            view.getEngine().getLoadWorker().stateProperty().addListener(new ChangeListener<State>() {
+                @Override
+                public void changed(ObservableValue<? extends State> ov, State t, State t1) {
+                    if(t1 == State.SUCCEEDED) {
+                        waitTestDocumentLoaded.countDown();
+                    }
+                }
+            });
```

Comment:
All these boilerplate code has been already taken care by TestBase, just derive it from TestBase and use it's helper methods. Refer SVGTest.java for more information.

`testClickOnStrokePointerEventsStroke` shall be simplified like below,
```
    @Test
    public void testClickOnStrokePointerEventsStroke() throws Exception {
        load(PointerEventsTest.class.getClassLoader().getResource("test/html/pointerevents-stroke.html").toExternalForm());
        submit(() -> {
            final WebPage page = WebEngineShim.getPage(getEngine());
            // Render WebPage so that all of the svg paths also will be rendered.
            WebPageShim.paint(page, 0, 0, 800, 600);

            WebPageShim.click(page, 130, 80);
            WebPageShim.click(page, 330, 80);
            WebPageShim.click(page, 530, 80);
            WebPageShim.click(page, 130, 280);
            WebPageShim.click(page, 330, 280);
            WebPageShim.click(page, 530, 280);

            for (String s : asList("polyline", "path", "rect", "circle", "ellipse", "polygon")) {
                getEngine().executeScript("document.getElementById('%s').click()".format(s));
                assertTrue("Expected element '" + s + "' to be activated", (boolean) getEngine().executeScript("isActivated('" + s + "')"));
            }
        });
    }
```